### PR TITLE
Support MSYS2/mingw-w64

### PR DESCRIPTION
**Pull request by mingwandroid**
Switching from $dayjob to $other-opensource-stuff .. this allows building on MSYS2/mingw-w64. I plan to submit a PR soon to MSYS2 for inclusion.

Thanks for your kind reviews!

### CODE REVIEW

**Comment by mingwandroid** [comment]
Oh, the failure we avoid here is:

```
/bin/sh ../libtool  --tag=CC --tag=disable-static  --mode=link gcc -I/c/opt/r/w/mingw-w64-frei0r-plugins/src/frei0r-plugins-1.7.0/include -Waddress -Wtype-limits -Wsign-compare -march=x86-64 -mtune=generic -O2 -pipe -module -avoid-version -lm -export-dynamic -pipe -Wl,--dynamicbase,--high-entropy-va,--nxcompat,--default-image-base-high -o alphainjection.la -rpath /mingw64/lib/frei0r-1 mixer2/alphainjection/alphainjection.lo
/bin/sh ../libtool  --tag=CXX --tag=disable-static  --mode=link g++  -march=x86-64 -mtune=generic -O2 -pipe -module -avoid-version -lm -export-dynamic -pipe -Wl,--dynamicbase,--high-entropy-va,--nxcompat,--default-image-base-high -o alphaout.la -rpath /mingw64/lib/frei0r-1 mixer2/alphaout/alphaout.lo
libtool:   error: can't build x86_64-w64-mingw32 shared library unless -no-undefined is specified
```

**Comment by ddennedy** [comment]
FYI, CMake works [for me](https://github.com/mltframework/shotcut/blob/master/scripts/build-shotcut-msys2.sh#L452) in msys2. I build

**Comment by mingwandroid** [comment]
Sure, that's great, but I'd like to use autotools personally!

**Comment by mingwandroid** [comment]
This isn't quite ready yet.

**Comment by mingwandroid** [comment]
I think this is ready for review now.

File: .github/workflows/ci-build.sh
Comment (ddennedy) [inline .github/workflows/ci-build.sh:11]: `ci-library.sh` is not available here

File: .github/workflows/ci-build.sh
Comment (ddennedy) [inline .github/workflows/ci-build.sh:13]: I do not think it is cool to be using msys2 e-mail here.

File: .github/workflows/ci-build.sh
Comment (ddennedy) [inline .github/workflows/ci-build.sh:31]: I am not sure what all this script is doing (referring to `define_build_order` and `check_recipe_quality`) even with `ci-library.sh` included, and I do not believe that we maintainers want to left to maintain it.

Comment (mingwandroid) [reply .github/workflows/ci-build.sh:31]: I am happy to drop this bit, unforunately, it can't be tested until it has been merged, which is a bit of a catch-22 situation anyway.

This is pacman/makepkg stuff. I used the canonical MSYS2 ci setup.

File: .github/workflows/ci-build.sh
Comment (mingwandroid) [reply .github/workflows/ci-build.sh:13]: Yeah, probably not a good idea!

File: .github/workflows/ci-build.sh
Comment (mingwandroid) [reply .github/workflows/ci-build.sh:11]: Good spot, thanks!

**Comment by mingwandroid** [comment]
Here is my PR to add frei0r to MSYS2: https://github.com/msys2/MINGW-packages/pull/7367

**Comment by lazka** [comment]
What's the point of adding our CI scripts here? Are there any PKGBUILDs?

**Comment by mingwandroid** [comment]
Thanks. I'll read up and fix it. I'll also split adding ci into a separate pr.

**Comment by mingwandroid** [comment]
Hi @ddennedy, I removed all the GitHub actions stuff from this PR and think the rest is ready for a re-review if you had time?

**Review by ddennedy — APPROVED** [review]

**Pull request merged**
